### PR TITLE
Populate instrument parameters in AddSampleLog

### PR DESCRIPTION
**Pull request by peterfpeterson**
The SNAP instrument has 4 log values that help define the position of its detectors. Currently one has to `AddSampleLog` then `LoadInstrument` in order to update the detector positions. This simplifies that workflow by adding a parameter to `AddSampleLog` that updates the instrument parameters.

*There is no associated issue,* but it is related to [EWM1842](https://ornlrse.clm.ibmcloud.com/ccm/web/projects/Neutron%20Data%20Project%20(Change%20Management)#action=com.ibm.team.workitem.viewWorkItem&id=1842)

### To test:

This is the prototype script that was written to help develop the new functionality.
```python
from mantid.simpleapi import mtd, LoadEmptyInstrument, AddSampleLog
from mantid.kernel import V3D
import numpy as np

positions = {'det_lin1':0.045, 'det_lin2':0.043,
             'det_arc1':-65.3, 'det_arc2':104.95}
wksp_name="SNAP_det_pos"

LoadEmptyInstrument(InstrumentName="SNAP", OutputWorkspace=wksp_name)
for name, value in positions.items():
    AddSampleLog(Workspace=wksp_name, LogName=name, LogText=str(value), LogType="Number Series", UpdateInstrumentParameters=True)

instr = mtd[wksp_name].getInstrument()

westPos = instr.getComponentByName('West').getPos()
assert np.rad2deg(westPos.angle(V3D(0,0,1))) == np.abs(positions['det_arc1']) # angle returns positive
assert westPos.distance(V3D(0,0,0)) == 0.5 + positions['det_lin1']

eastPos = instr.getComponentByName('East').getPos()
assert np.rad2deg(eastPos.angle(V3D(0,0,1))) == positions['det_arc2']
assert eastPos.distance(V3D(0,0,0)) == 0.5 + positions['det_lin2']
```

---

### Reviewer

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

#### Code Review

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Do the release notes conform to the [release notes guide](https://developer.mantidproject.org/Standards/ReleaseNotesGuide.html)?

#### Functional Tests

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### Gatekeeper

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

### CODE REVIEW

File: Framework/Algorithms/test/AddSampleLogTest.h [hunk 1]
```diff
@@ -21,6 +23,11 @@ using namespace Mantid::Algorithms;
 using Mantid::Types::Core::DateAndTime;
 
 class AddSampleLogTest : public CxxTest::TestSuite {
+private:
+  void setStarEndTime(MatrixWorkspace_sptr &ws) {
```
[the remark this answers is not in the document]
Comment (peterfpeterson) [reply Framework/Algorithms/test/AddSampleLogTest.h:27]: The function `AddSampleLogTest::ExecuteAlgorithm()` assumes that the start and end time for the run are set and the logs will pick them up. I saw the same strange call 4 times in the test file so I refactored it into one call.

[the remark this answers is not in the document]
Comment (rboston628) [reply Framework/Algorithms/test/AddSampleLogTest.h:27]: I think you misspelled it

**Comment by peterfpeterson** [comment]
I rebased on `ornl-next` with the intention of making the diff smaller and it picked up even **more** changes. :-1:

**Review by rboston628 — DISMISSED** [review]

**Review by rboston628 — DISMISSED** [review]

**Review by rboston628 — DISMISSED** [review]

**Review by rboston628 — DISMISSED** [review]

**Review by rboston628 — APPROVED** [review]

**Review by rboston628 — DISMISSED** [review]

**Review by rboston628 — DISMISSED** [review]

**Review by darshdinger — DISMISSED** [review]

**Review by darshdinger — APPROVED** [review]
This looks good...

**Review by rboston628 — APPROVED** [review]
I approve

**Comment by rboston628** [comment]
Test

**Review by rboston628 — APPROVED** [review]

**Review by rboston628 — DISMISSED** [review]

**Review by darshdinger — DISMISSED** [review]
Alt Text

**Review by rboston628 — DISMISSED** [review]

**Comment by peterfpeterson** [comment]
This has been replaced with #36539 which is actually allowing itself to be reviewed

**Pull request closed without merging**